src/sympc/module/nn/conv.py

```diff
@@ -74,20 +75,36 @@ def forward(self, x: MPCTensor) -> MPCTensor:
 
     __call__ = forward
 
-    def share_state_dict(self, state_dict: Dict[str, Any]) -> None:
+    def share_state_dict(
+        self,
+        state_dict: Dict[str, Any],
+    ) -> None:
         """Share the parameters of the normal Conv2d layer.
 
         Args:
             state_dict (Dict[str, Any]): the state dict that would be shared
         """
+
+        bias = None
+        if ispointer(state_dict):
+            weight = state_dict["weight"].resolve_pointer_type()
```

Comment:
sorry for the stupid Q: what does `resolve_pointer_type` do?

Reply:
state_dict is an ```OrderedDictPointer``` and doing a ```get``` on it we do not know what we get back (we get back a generic type named ```AnyPointer```). To be able to call ```shape``` we need to solve its type (```resolve_pointer_type``` would give back a ```TensorPointer```)

Reply:
Hmmm I see :)